### PR TITLE
Add s3 policy + instance role

### PR DESCRIPTION
**Pull request by azizova-leyla**
File upload still doesn't work properly, but s3 setup looks correct.

https://github.com/seattle-uat/civiform/issues/2554
https://github.com/seattle-uat/civiform/issues/2511

### CODE REVIEW

**Review by bion — CHANGES_REQUESTED** [review]
probably shouldn't check in AWSCLIV2.pkg

**Comment by azizova-leyla** [comment]
> probably shouldn't check in AWSCLIV2.pkg

definitely not =)

File: cloud/aws/templates/aws_oidc/aim.tf [hunk 1]
```diff
@@ -0,0 +1,34 @@
+resource "aws_iam_role" "apprunner_instance_role" {
```
Comment (goldblatt) [inline cloud/aws/templates/aws_oidc/aim.tf:1]: why is this here instead of the aws directory

File: cloud/aws/templates/aws_oidc/main.tf [hunk 2]
```diff
@@ -28,10 +28,8 @@ resource "aws_apprunner_service" "civiform_dev" {
           WHITELABEL_LOGO_WITH_NAME_URL      = var.civic_entity_logo_with_name_url
           SUPPORT_EMAIL_ADDRESS              = var.civic_entity_support_email_address
 
-          AWS_SES_SENDER        = var.ses_sender_email
-          AWS_ACCESS_KEY_ID     = "CHANGE_ME"
-          AWS_SECRET_ACCESS_KEY = "CHANGE_ME"
-          AWS_REGION            = var.aws_region
+          AWS_SES_SENDER = var.ses_sender_email
+          AWS_REGION     = var.aws_region
```
Comment (goldblatt) [inline cloud/aws/templates/aws_oidc/main.tf:32]: why don't these get passed in? it might complain if they aren't at least dummy?

File: cloud/aws/templates/aws_oidc/main.tf [hunk 3]
```diff
@@ -92,14 +93,36 @@ resource "aws_s3_bucket" "civiform_files_s3" {
...
+    actions   = ["s3:*"]
+    effect    = "Deny"
+    resources = ["${aws_s3_bucket.civiform_files_s3.arn}/*"]
+    not_principals {
+      type        = "AWS"
+      identifiers = [aws_iam_role.apprunner_instance_role.arn]
+    }
+  }
+  statement {
+    actions   = ["s3:GetObject", "s3:PutObject"]
```
Comment (goldblatt) [inline cloud/aws/templates/aws_oidc/main.tf:120]: ooo is this like you can get but you need to have authenticated first?

File: cloud/aws/templates/aws_oidc/main.tf [hunk 4]
```diff
@@ -92,14 +93,36 @@ resource "aws_s3_bucket" "civiform_files_s3" {
   bucket = "civiform-files-s3"
 }
 
-resource "aws_s3_access_point" "civiform_files" {
+resource "aws_s3_bucket_public_access_block" "civiform_files_access" {
```
Comment (goldblatt) [inline cloud/aws/templates/aws_oidc/main.tf:96]: i also think these should go in the generic aws folder and not the template folder

**Review by goldblatt — CHANGES_REQUESTED** [review]

File: cloud/aws/templates/aws_oidc/main.tf (hunk 3 above)
Comment (azizova-leyla) [reply cloud/aws/templates/aws_oidc/main.tf:120]: yup. only the apprunner job should have access to this. Seattle does it in a similar way.

File: cloud/aws/templates/aws_oidc/aim.tf (hunk 1 above)
Comment (azizova-leyla) [reply cloud/aws/templates/aws_oidc/aim.tf:1]: added todo at the top.

**Comment by bion** [comment]
nice, is file upload working now on staging?

**Review by bion — APPROVED** [review]

**Comment by azizova-leyla** [comment]
> 


> nice, is file upload working now on staging?
Not yet, there are more things that need to be fixed for that.

**Review by goldblatt — APPROVED** [review]
